### ISSUE TITLE
Set $id after clone

### ISSUE THREAD

**Issue by Eomm** [fastify/fluent-json-schema#193]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

Given an origin flunet-json-schema object filtered with `only/without` it is not possible to set the BaseSchema's `$id`:

```
const WhizBang = fluent
  .object()
  .additionalProperties(false)
  .id('whizbang')
  .title('WhizBang')
  .prop(
    'whiz'
  , fluent.string().required().minLength(1)
  )
  .prop(
    'bang'
  , fluent.string().required().minLength(1)
  )

const part = WhizBang.only(['whiz'])
console.log(JSON.stringify(part.id('asd').valueOf(), null, 2))
```

Shows:

```diff
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "object",
  "additionalProperties": false,
  "$id": "whizbang",
  "title": "WhizBang",
  "properties": {
    "whiz": {
+      "$id": "asd",
      "type": "string",
      "minLength": 1
    }
  },
  "required": [
    "whiz"
  ]
}
```

I can't find a way to set the `"$id": "whizbang"` field.

Ref https://github.com/fastify/fastify/issues/4276

**Comment by zrosenbauer** [comment]
@Eomm instead of resetting the ID would it make sense to just omit it? I'm not sure it should even reference the $id at this point...

.only & .without are state mutations that invalidate the ref aka should compose new objects not mutate the old?

**Comment by esatterwhite** [comment]
It would need to be both. It would need to omit the id by default, and allow you to set it in the correct location of the schema

**Comment by zrosenbauer** [comment]
I'm happy to open up a PR handling both if @Eomm or other core folks (@fastify/plugins) thinks it make sense.

**Comment by Eomm** [comment]
Yes, I agree:

- remove it by default
- allow to set it again by calling `.id()`

Note that this would be a breaking change

**Comment by esatterwhite** [comment]
The problem is rooted in the use of `setAttribute` of BaseSchema

https://github.com/fastify/fluent-json-schema/blob/490c24d577221038c21403564c8d0248466f7570/src/utils.js#L188-L195

Once an object has properties, all things are added to its properties rather than the root of the schema. this doesn't seem too hard to fix

**Comment by aboutlo** [comment]
resolved by #195
